### ISSUE TITLE
Compass rotation changes position of note draft

### ISSUE THREAD

**Issue by ltog** [streetcomplete/StreetComplete#3101]
Opening a draft for a note, then rotating the map, changes the position where the map note would be positioned.

**How to Reproduce**
1. Change the compass direction of the map by about 90 degrees away from north using a two-finger rotational gesture
2. Long press on the top half of the map on a distinctive place, e.g. a crossroad.
3. Choose `Create new note`
4. Notice the position of the `!` marker
5. Press the compass symbol (thereby getting north at the top)
6. Notice how the `!` symbol changed its position :-( I would expect instead that the marker still points to the same location on the map.

**Versions affected**

StreetComplete v33.1 on Android 10

**Comment by westnordost** [comment]
This is intended / will not fix.

It is intended that you can move the map to adjust the position of the note. This is also explained in the text above the message when adding a note.
Rotating the map still falls under adjusting the map position. If you want the marker to stay at the center of the rotation, then you need to place your fingers so that the marker is at the center of rotation. It is a will not fix to change the behavior of the rotation-gesture to automatically center the rotation on the marker (because it is technically probably not possible with tangram-es and also not worth implementing it if it were, IMO)

**Comment by smichel17** [comment]
I think you've misread. This is about auto-rotation back to north when tapping the compass icon.

**Comment by smichel17** [comment]
Okay, I tried it. I think this is solve-able. I'll look into it within the next couple days.

**Comment by ltog** [comment]
Yes @smichel17, this is about auto-rotating using the compass-icon, NOT about manually rotating using a two-finger gesture.

Thanks for looking into it!

**Comment by westnordost** [comment]
Ah I see. Thanks, @smichel17 for giving it a try.

**Comment by smichel17** [comment]
Okay, so this is slightly more complicated than I thought.

We do indeed have all the right information we need.

- The `MapFragment` has utility functions for working with coordinates vs screen offsets
   https://github.com/streetcomplete/StreetComplete/blob/6b86eb09260f06858dd927e56c28939dbfe0961e/app/src/main/java/de/westnordost/streetcomplete/map/MapFragment.kt#L329-L369

- The `MainFragment` stores the offsets of the current open dialog
   https://github.com/streetcomplete/StreetComplete/blob/6b86eb09260f06858dd927e56c28939dbfe0961e/app/src/main/java/de/westnordost/streetcomplete/map/MainFragment.kt#L111
   And it has a callback for responding to compass taps (we'd change the last `else`, line `611`)
   https://github.com/streetcomplete/StreetComplete/blob/6b86eb09260f06858dd927e56c28939dbfe0961e/app/src/main/java/de/westnordost/streetcomplete/map/MainFragment.kt#L591-L618

However, I don't actually see how to make Tangram rotate around a point on the map that's not the center, as if you were doing it with your fingers. I can, at the last, make it end in the correct position; it just might look a little funky. I'll play around and see how awkward it is, and in the meantime, if anybody wants to trawl through the **[tangram docs](https://tangrams.readthedocs.io/en/master/API-Reference/android-sdk/0.11.0/index.html)** and see if there's a way to affect the center rotation coordinates, that would be nice :)

**Comment by westnordost** [comment]
Thank you for your effort, @smichel17 

If the tangram SDK does not provide a method for that, I would then suggest to close this issue as will not fix, optionally create a feature request at tangram-es. We shouldn't put so much effort into trying to workaround missing features in tangram-es because in the long- or mid-term, we should think about replacing tangram-es for example with the MapLibre SDK
 , also in view of a possible iOS port (https://github.com/streetcomplete/StreetComplete/issues/1892#issuecomment-889991925). Using a different map renderer will solve a whole lot of issues (but create others probably)

**Comment by smichel17** [comment]
I opened https://github.com/tangrams/tangram-es/issues/2284; let's open a new issue to track MapLibre exploration. I'll do this later; I am about to travel.

**Comment by smichel17** [comment]
Upstream issue was fixed, so this will be possible to implement with the next Tangram release.

**Comment by westnordost** [comment]
The only reason why this is not set as will not fix is because smichel wanted to work on it. Now that he does not work on it anymore, I'll close this. However, anyone reading this who would like to look into it: Go ahead.
